src/js/App/Feedback/ReportBug.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useState } from 'react';
+import {
+  Button,
+  Checkbox,
+  Form,
+  FormGroup,
+  Panel,
+  PanelMain,
+  PanelMainBody,
+  Text,
+  TextArea,
+  TextContent,
+  TextVariants,
+} from '@patternfly/react-core';
+import { ExternalLinkAltIcon } from '@patternfly/react-icons';
+import { DeepRequired } from 'utility-types';
+import { ChromeUser } from '@redhat-cloud-services/types';
+import './Feedback.scss';
+import PropTypes from 'prop-types';
+import { getEnv, getUrl, isProd } from '../../utils';
+import { useIntl } from 'react-intl';
+import messages from '../../Messages';
+
+export type ReportBugProps = {
+  user: DeepRequired<ChromeUser>;
+  onCloseModal: () => void;
+  onSubmit: () => void;
+  onClickBack: () => void;
+  handleFeedbackError: () => void;
+};
+
+const ReportBug = ({ user, onCloseModal, onSubmit, onClickBack, handleFeedbackError }: ReportBugProps) => {
+  const [textAreaValue, setTextAreaValue] = useState('');
+  const [checked, setChecked] = useState(false);
+  const intl = useIntl();
+  const env = getEnv();
+  const app = getUrl('app');
+  const bundle = getUrl('bundle');
+  const isAvailable = env === 'prod' || env === 'stage';
+  const addFeedbackTag = () => (isProd() ? `[${bundle}]` : '[PRE-PROD]');
+
+  async function handleModalSubmission() {
+    const token = await window.insights.chrome.auth.getToken();
+    if (isAvailable) {
+      try {
+        const response = await fetch(`${window.origin}/api/platform-feedback/v1/issues`, {
```

Comment:
The fetch function seems to be always almost the same, only the body changes. Can we create one general function and configure it with arguments?

Reply:
I actually further refactored it